WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs

```diff
@@ -318,6 +318,14 @@ public async Task SignTransactionAsync(TransactionSignaturesRequest request, Can
 		{
 			var round = GetRound(request.RoundId, Phase.TransactionSigning);
 
+			if (round.Parameters.ForceSigningDelay)
+			{
+				if (!round.TransactionSigningDelayTimeFrame.HasExpired)
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.SignatureTooSoon, $"Round ({round.Id}): Received signature before transaction signing delay has expired.");
+				}
+			}
```

Comment:
I would reduce the nesting here:

```suggestion
			if (round.Parameters.ForceSigningDelay && !round.TransactionSigningDelayTimeFrame.HasExpired)
			{
					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.SignatureTooSoon, $"Round ({round.Id}): Received signature before transaction signing delay has expired.");
			}
```